### PR TITLE
Adds index_location to sharding codec

### PR DESCRIPTION
**Pull request by normanrz**
As discussed in #254 and proposed by @jbms, this PR adds the `index_location` configuration parameter to the `sharding_indexed` codec.

Allowing to place the shard index at the beginning of a file may be beneficial for writing shard on stores that support random writes.

### CODE REVIEW

File: docs/v3/codecs/sharding-indexed/v1.0.rst [hunk 1]
```diff
@@ -155,6 +156,11 @@ Sharding can be configured per array in the :ref:`array-metadata` as follows::
     be used for index codecs. It is RECOMMENDED to use a little-endian codec 
     followed by a crc32c checksum as index codecs.
 
+``index_location``
```
Comment (jbms) [inline docs/v3/codecs/sharding-indexed/v1.0.rst:159]: For backwards compatibility with existing data we should say that if not present, defaults to "end".

File: docs/v3/codecs/sharding-indexed/v1.0.rst [hunk 2]
```diff
@@ -25,12 +25,12 @@ is licensed under a `Creative Commons Attribution 3.0 Unported License
...
 
-.. rst-class:: draft
+ZEP0002 was accepted on November 1st, 2023 via https://github.com/zarr-developers/zarr-specs/issues/254.
 
 .. warning::
-    This document is a draft for review and subject to changes.
+    This document is subject to changes.
     It will become final when the `Zarr Enhancement Proposal (ZEP) 2 <https://zarr.dev/zeps/draft/ZEP0002.html>`_
-    is approved via the `ZEP process <https://zarr.dev/zeps/active/ZEP0000.html>`_.
+    is finalized via the `ZEP process <https://zarr.dev/zeps/active/ZEP0000.html>`_.
```
Comment (joshmoore) [inline docs/v3/codecs/sharding-indexed/v1.0.rst:33]: Aren't there still a few more issues/prs though? :smile:

Comment (normanrz) [reply docs/v3/codecs/sharding-indexed/v1.0.rst:33]: Doesn't it say that it *will* become final when finalized?

**Review by jbms — APPROVED** [review]

**Comment by MSanKeys963** [comment]
Should we move ahead with this @normanrz?

**Comment by normanrz** [comment]
Yes! I don't have merge rights so please go ahead and merge.

**Pull request merged**